core/src/com/unciv/logic/civilization/CivInfoStats.kt

```diff
@@ -185,6 +196,19 @@ class CivInfoStats(val civInfo: CivilizationInfo) {
 
         statMap["Natural Wonders"] = happinessPerNaturalWonder * civInfo.naturalWonders.size
 
+        if (civInfo.religionManager.religion != null) {
+            statMap["Religion"] = 0f
+            for (unique in civInfo.religionManager.religion!!.getFounderBeliefs().flatMap { it.uniqueObjects }) {
+                if (unique.placeholderText == "[] for each global city following this religion") {
+                    statMap["Religion"] = statMap["Religion"]!! + unique.stats.apply {
+                        this.timesInPlace(civInfo.religionManager.numberOfCitiesFollowingThisReligion().toFloat())
+                    }.happiness
+                }
+            }
+            if (statMap["Religion"] == 0f) 
+                statMap.remove("Religion")
```

Comment:
`add - if() remove` pattern instead of `if() add` -> -0.001 ❤️. Hmmm... We need a moddable 'currency' named 'Love' in Unciv? Sorry. A little more seriously: I need to go fetch my Studio, from that glance it shouldn't compile? Isn't StatMap[key] a `Stats` and why then can we assign/compare with a Float?

Reply:
You would think that, but in this instance it is a map mapping 'origins' to 'happiness' values, as this function only computes happiness, which needs to be computed separately so that effects like "+15% science if the empire is happy" can be resolved correctly. And while I could test for it beforehand, that would require doing `getMatchingUniques()` twice, which would incur more work, and if in the end the values just happen to sum up to 0 it is still displayed which you also don't want. This seemed to be the best solution to me

Reply:
Riiight, with enough code in view it's plain. I just got confused as there is a class named StatMap.